### ISSUE TITLE
Error handling Field Filters on PostgreSQL schemas with hyphen

### ISSUE THREAD

**Issue by dgnunes** [metabase/metabase#8766]
### Problem

I work on an environment where schemas are composed of two UUIDs (like "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"). When I use Field Filters from an SQL Question the following sample query:

SELECT SUM(value), "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FORMASPG"."desc" FROM "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FINANPAG" F INNER JOIN "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FORMASPG" ON F.code = "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FORMASPG"."code" WHERE {{descrip}}

Becomes this on the DB:

SELECT SUM(value), "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FORMASPG"."desc" FROM "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FINANPAG" F INNER JOIN "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FORMASPG" ON F.code = "207b071c-eb9c-4d0f-b84a-9_e4136efa-b92b-40d0-b960-08147e0acd52"."FORMASPG"."code" WHERE "207b071c_eb9c_4d0f_b84a_9_e4136efa_b92b_40d0_b960_08147e0acd52"."FORMASPG"."desc" = $1

The same error happens with different field filters types (Dates, Text and others).

Tl;dr

All hyphen in schema name turned into underscores when metabase handles field filters.

### Specs
- Browse : Firefox 62.0.3 
- DBEngine : PostgreSQL 10.4  
- Metabase version: v0.30.4

**Comment by camsaul** [comment]
@dgnunes ?

**Comment by camsaul** [comment]
Confirmed fixed by #9198
